data/Record.js

```diff
@@ -146,10 +181,17 @@ export class Record {
             this.parentId == rec.parentId &&
             equal(this.xhTreePath, rec.xhTreePath) &&
             this.store == rec.store &&
-            this.store.fields.every(f => f.isEqual(this[f.name], rec[f.name]))
+            equal(this.data, rec.data)
         );
     }
 
+    isFieldDirty(name) {
+        if (!this.isDirty) return false;
+
+        const value = this.data[name], originalValue = this.originalRecord.data[name];
```

Comment:
could we use getters on record here?  also, why are removing the field.isEqual check?